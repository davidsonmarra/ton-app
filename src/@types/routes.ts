export type ProductsStackParamList = {
  HomeScreen: undefined;
  ProductDetailsScreen: { id: number };
};
