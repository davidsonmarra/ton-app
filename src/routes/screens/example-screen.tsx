import React from 'react';
import { Example } from '../../presentational';

export const ExampleScreen = () => <Example />;
