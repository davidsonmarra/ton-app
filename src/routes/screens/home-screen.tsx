import React from 'react';
import { Home } from '../../presentational';

export const HomeScreen = () => <Home />;
