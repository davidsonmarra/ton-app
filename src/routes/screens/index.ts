export { ExampleScreen } from './example-screen';
export { ProductDetailsScreen } from './product-details-screen';
export { ProductsListScreen } from './products-list-screen';
