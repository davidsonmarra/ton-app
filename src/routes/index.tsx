import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { ProductsRoutes } from './stacks/products.routes';

export function Routes() {
  return (
    <NavigationContainer>
      <ProductsRoutes />
    </NavigationContainer>
  );
}
