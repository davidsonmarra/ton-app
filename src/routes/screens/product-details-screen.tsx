import React, { useCallback, useLayoutEffect, useState } from 'react';
import { ProductDetailsDTO, ProductsStackParamList } from '../../@types';
import { ProductDetails } from '../../presentational';
import { NavigationProp, RouteProp, useNavigation, useRoute } from '@react-navigation/native';

export const ProductDetailsScreen = () => {
  const [product, setProduct] = useState({} as ProductDetailsDTO);
  const [isLoading, setIsLoading] = useState(true);

  const { goBack } =
    useNavigation<NavigationProp<ProductsStackParamList, 'ProductDetailsScreen'>>();
  const { id } = useRoute<RouteProp<ProductsStackParamList, 'ProductDetailsScreen'>>().params;

  const getProduct = useCallback(async () => {
    const response = await fetch('api/products/' + id);
    const data = await response.json();
    setProduct(data);
    setIsLoading(false);
  }, [product]);

  const handlePressLeftIcon = useCallback(() => {
    goBack();
  }, []);

  useLayoutEffect(() => {
    getProduct();
  }, []);

  return (
    <ProductDetails
      isLoadingApi={isLoading}
      item={product}
      handlePressLeftIcon={handlePressLeftIcon}
    />
  );
};
