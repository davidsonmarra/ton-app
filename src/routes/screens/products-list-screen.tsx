import React, { useCallback, useLayoutEffect, useState } from 'react';
import { ProductsList } from '../../presentational';
import { ProductDTO, ProductsStackParamList } from '../../@types';
import { NavigationProp, useNavigation } from '@react-navigation/native';
import { api } from '../../global/services';

export const ProductsListScreen = () => {
  const [products, setProducts] = useState<ProductDTO[]>([]);
  const [infoModalIsVisible, setInfoModalIsVisible] = useState(false);
  const { navigate } =
    useNavigation<NavigationProp<ProductsStackParamList, 'ProductsListScreen'>>();

  const getProducts = useCallback(async () => {
    try {
      const { data } = await api.get('products');
      setProducts(data.products);
    } catch (error) {
      console.log(error);
    }
  }, []);

  const handlePressProduct = useCallback((id: number) => {
    navigate('ProductDetailsScreen', { id });
  }, []);

  const toogleModal = useCallback(() => {
    setInfoModalIsVisible(!infoModalIsVisible);
  }, [infoModalIsVisible]);

  useLayoutEffect(() => {
    getProducts();
  }, []);

  return <ProductsList products={products} handlePressProduct={handlePressProduct} infoModalIsVisible={infoModalIsVisible} toogleModal={toogleModal} />;
};
