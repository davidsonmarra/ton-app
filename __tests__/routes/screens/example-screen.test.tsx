import { ExampleScreen } from '../../../src/routes/screens';
import { matchSnapshotWithProviders, render, act } from '../../test-utils';


describe('ProductDetailsScreen', () => {
  it('should match a snapshot', async () => {
    await act(() => {
      matchSnapshotWithProviders(<ExampleScreen />);
    });
  });

  it('should render correctly', () => {
      const { getByText } = render(<ExampleScreen />);
      expect(getByText('PÃ¡gina de Exemplo')).toBeTruthy();
  });
});
