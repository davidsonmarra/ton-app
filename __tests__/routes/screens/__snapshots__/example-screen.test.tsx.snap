// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailsScreen should match a snapshot 1`] = `null`;
